hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AbstractFSWAL.java

```diff
@@ -2022,13 +2023,23 @@ protected final void waitForSafePoint() {
     }
   }
 
+  private void recoverLease(FileSystem fs, Path p, Configuration conf) {
+    try {
+      RecoverLeaseFSUtils.recoverFileLease(fs, p, conf, null);
+    } catch (IOException ex) {
+      LOG.error("Unable to recover lease after several attempts. Give up.", ex);
+      throw new RuntimeException(ex);
```

Comment:
Better not do this...

We are running in a background thread pool and this will kill the thread in the thread pool...
If the thread pool can still create new threads then we are safe, but it is useless to throw the exception out here. And if the thread pool is broken, then we are in trouble...

Reply:
Good point. How about just log the exception and move on?